core/src/test/java/com/sequenceiq/cloudbreak/service/upgrade/sync/CmSyncerServiceTest.java

```diff
@@ -1,118 +1,111 @@
-        verify(cmInstalledComponentFinderService).findParcelComponents(eq(stack), eq(candidateImages));
-
-        ArgumentCaptor<Set<Component>> componentArgumentCaptor = ArgumentCaptor.forClass(Set.class);
-        verify(stackComponentUpdater).updateComponentsByStackId(eq(stack), componentArgumentCaptor.capture(), anyBoolean());
-        Set<Component> componentsToPersist = componentArgumentCaptor.getValue();
-        assertThat(componentsToPersist, hasSize(2));
-        assertThat(componentsToPersist, containsInAnyOrder(
-                hasProperty("componentType", is(ComponentType.CDH_PRODUCT_DETAILS)),
-                hasProperty("componentType", is(ComponentType.CM_REPO_DETAILS))
-        ));
-
-        ArgumentCaptor<Set<Component>> clusterComponentArgumentCaptor = ArgumentCaptor.forClass(Set.class);
-        verify(clusterComponentUpdater).updateClusterComponentsByStackId(eq(stack), clusterComponentArgumentCaptor.capture(), anyBoolean());
-        Set<Component> clusterComponentsToPersist = clusterComponentArgumentCaptor.getValue();
-        assertThat(clusterComponentsToPersist, hasSize(2));
-        assertThat(componentsToPersist, containsInAnyOrder(
-                hasProperty("componentType", is(ComponentType.CDH_PRODUCT_DETAILS)),
-                hasProperty("componentType", is(ComponentType.CM_REPO_DETAILS))
-        ));
+        CmRepoSyncOperationResult cmRepoSyncOperationResult = mock(CmRepoSyncOperationResult.class);
+        CmParcelSyncOperationResult cmParcelSyncOperationResult = mock(CmParcelSyncOperationResult.class);
+        when(cmInstalledComponentFinderService.findCmRepoComponent(stack, candidateImages)).thenReturn(cmRepoSyncOperationResult);
+        when(cmInstalledComponentFinderService.findParcelComponents(stack, candidateImages)).thenReturn(cmParcelSyncOperationResult);
+        CmSyncOperationResult cmSyncOperationResult = new CmSyncOperationResult(cmRepoSyncOperationResult, cmParcelSyncOperationResult);
+        when(cmSyncOperationResultFactory.create(cmRepoSyncOperationResult, cmParcelSyncOperationResult)).thenReturn(cmSyncOperationResult);
+        when(cmSyncOperationSummaryService.evaluate(cmSyncOperationResult)).thenReturn(CmSyncOperationSummary.builder().withSuccess("myMessage").build());
+
+        CmSyncOperationSummary cmSyncOperationSummary = underTest.syncFromCmToDb(stack, candidateImages);
+
+        assertTrue(cmSyncOperationSummary.hasSucceeded());
+        assertEquals("myMessage", cmSyncOperationSummary.getMessage());
+        verify(cmInstalledComponentFinderService).findCmRepoComponent(stack, candidateImages);
+        verify(cmInstalledComponentFinderService).findParcelComponents(stack, candidateImages);
+        verify(cmSyncOperationSummaryService).evaluate(cmSyncOperationResult);
+        verify(cmSyncOperationResultFactory).create(cmRepoSyncOperationResult, cmParcelSyncOperationResult);
```

Comment:
instead of verifying this, and creating a service for this reason, you should use argumentcaptor for `cmSyncOperationSummaryService#evaluate` and `componentPersistingService#persistComponentsToDb` parameter and verify they are called with a `CmSyncOperationResult` which has the right values. So you could drop `cmSyncOperationResultFactory` which is basically replacing the constructor

Reply:
Yes, that's also an option. When possible I like to avoid argumentCaptors, as they make a test way longer and harder to read. But yes, I can add.

Reply:
Done